test/extended/util/disruption/disruption.go

```diff
@@ -325,6 +325,22 @@ func ExpectNoDisruption(f *framework.Framework, tolerate float64, total time.Dur
 	}
 }
 
+// ExpectNoDisruptionForDuration fails if the sum of the duration of all events exceeds allowedDisruption, reports a
+// disruption flake if any disruption occurs, and uses reason to prefix the message.
+func ExpectNoDisruptionForDuration(f *framework.Framework, allowedDisruption time.Duration, total time.Duration, events monitorapi.Intervals, reason string) {
```

Comment:
This sets us up nicely for a future P95 lookup table based on release, platform, network type, and topology.